esphome/components/api/__init__.py

```diff
@@ -320,6 +350,23 @@ async def homeassistant_service_to_code(config, action_id, template_arg, args):
     for key, value in config[CONF_VARIABLES].items():
         templ = await cg.templatable(value, args, None)
         cg.add(var.add_variable(key, templ))
+
+    if response_template := config.get(CONF_RESPONSE_TEMPLATE):
+        templ = await cg.templatable(response_template, args, cg.std_string)
+        cg.add(var.set_response_template(templ))
+
+    if on_response := config.get(CONF_ON_RESPONSE):
```

Comment:
```suggestion
    if on_response := config.get(CONF_ON_RESPONSE):
        cg.add_define("USE_API_HOMEASSISTANT_ACTION_RESPONSES")
```

Since this is so big maybe conditionally compile response support?